Shopify/cli
# Refactor service environments

### WHY are these changes introduced?
CLI supported different environments for each service, but that doesn't really work. You can't use a production identity token with a spin/local service and vice versa. So it makes sense to define all services under the same environment always.





### WHAT is this pull request doing?
- Force all services to use the same environment (prod/local/spin)
- Remove extra env vars and create a new one for this: `SHOPIFY_SERVICE_ENV`
- Update the `spin` script with the new env vars
- SPIN=1 should not be manually used, should be used to detect spin environments.



### How to test your changes?
- `dev fixture dev` works normally
- Having a spin instance online: `dev spin yarn shopify... ` should use spin for everything
- Within a spin instance, `dev fixture dev` should use spin services



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
